File: 01_arrays_and_hashing/03_two_sum/solution.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}

        for i, x in enumerate(nums):
            x_target = target - x

            if x_target not in seen:
                seen[x] = i
            else:
                return [seen[x_target], i]

File: 01_arrays_and_hashing/03_two_sum/solution.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	hash := make(map[int]int)

	for index, num := range nums {
		numTarget := target - num

		if seenIdx, isPresent := hash[numTarget]; isPresent {
			return []int{seenIdx, index}
		}

		hash[num] = index
	}

	return []int{0, 0}
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9

	result := twoSum(nums, target)
	fmt.Println(result)
}
